sdk/storage/azdatalake/directory/client.go

```diff
@@ -227,6 +227,28 @@ func (d *Client) NewFileClient(fileName string) (*file.Client, error) {
 	return (*file.Client)(base.NewPathClient(fileURL, newBlobURL, newBlobClient, d.generatedDirClientWithDFS().InternalClient().WithClientName(shared.FileClient), d.sharedKey(), d.identityCredential(), d.getClientOptions())), nil
 }
 
+func (d *Client) NewSubdirectoryClient(subDirectoryName string) (*Client, error) {
```

Comment:
Let's add some tests that get a subdirectory client - check the URL, do some API calls.

Reply:
Accidentally pushed without the test I wrote!